### ISSUE TITLE
btm's hotkey cannot work in mintty

### ISSUE THREAD

**Issue by Freed-Wu** [ClementTsang/bottom#747]
Before you report, please take a look at [Troubleshooting](https://clementtsang.github.io/bottom/nightly/troubleshooting) to see if there's already an answer to your problem. Also check out [existing issues](https://github.com/ClementTsang/bottom/issues) and [known problems](https://clementtsang.github.io/bottom/nightly/support/official/#known-problems) to see if your problem is already reported/known/fixed.

Also, **please fill in all fields if possible** - if the issue is too hard to reproduce or vague, it may not be fixed!

## Describe the bug

Give a high level description of the bug.

In mintty, btm's hotkey cannot work.

## What operating system/architecture are you using?

Please mention what operating system, version, and architecture you're experiencing the problem on (ex: macOS 10.13, x86_64). Please also ensure that your system is [on the list of supported systems](https://github.com/ClementTsang/bottom#support) - bug reports on unsupported systems are likely to be closed.

In msys2,

```
❯ uname -a
MINGW64_NT-10.0-19044 windows10 3.3.3-341.x86_64 2021-12-30 06:18 UTC x86_64 Msys
❯ has btm
✓ btm 0.6.8
```

## What terminal are you using `bottom` on?

Please mention what terminal/terminal emulator you are using `bottom` on (ex: Konsole, Kitty), as well as their version and any relevant settings.

The bug only exists in `mintty`. Maybe this is due to mintty, like <https://github.com/dundee/gdu/issues/144>

## What version are you on?

Please mention which version of bottom you're running (`btm -V`, nightly, etc.)!

## How did you install `bottom`?

Please mention how you installed `bottom`. If you manually compiled it, please also mention your _Rust version_.

**Note: if you installed from cargo, ensure you installed the right crate (https://crates.io/crates/bottom).**

```
pacman -S mingw-w64-bottom
```

## How can we reproduce this?

Provide detailed steps on _how_ to reproduce your problem, to the best of your ability. Be as detailed as possible. Include config files or flags if relevant. If maintainers cannot reproduce the bug, it will be very hard to fix it.

## What is the actual behaviour?

What is the _actual_ result of the given steps? Be detailed. Please include screenshots if relevant and possible.

After `btm`, whatever press any key, nothing happened.

## What is the expected behaviour?

What do you believe the expected behaviour should be given these steps?

Hotkey should work.

**Comment by Freed-Wu** [comment]
See <https://github.com/msys2/MINGW-packages/pull/11662#issuecomment-1146853457>, use a shell script to wrap it can avoid this bug. (Although it is a little troublesome.)

**Comment by ClementTsang** [comment]
Nice that you got it resolved!

I'll add this to the documentation later for future reference if anyone has a similar issue.

**Comment by Freed-Wu** [comment]
Note winpty only work when stdin and stdout is a tty, such as `winpty XXX`, not `winpty XXX | YYY`. See `https://github.com/rprichard/winpty/issues/101`. So it is still not a perfect solution.

**Comment by Freed-Wu** [comment]
I found for that programs which installed by `pacman -S mingw-w64-XXX` in msys2 (XXX is gdu, fzf, exa, lsd, etc, which use fullscreen/ display color ), Window Terminal/msys2 shell can be a good terminal to display correctly. Such as:

In mintty/msys shell, `lsd --color auto`, `exa --color auto` cannot display correct color, but windows terminal/msys shell can.

In mintty/msys shell, `gdu`, `fzf` cannot display in fullscreen, but windows terminal/msys shell can.

In mintty/msys shell, `btm` cannot display in fullscreen, but windows terminal/msys shell can. However, for `btm`, it exists other problem. Although it can display in fullscreen, when users move cursor by `j/k`, the incorrect line will occur:

![Screenshot from 2022-08-10 15-28-14](https://user-images.githubusercontent.com/32936898/183842688-74d4e124-101b-450c-a30b-ed8d06c0296f.png)

Even if users use `winpty btm`, it also has incorrect problem. the cursor will be moved to the region which changed, Such as the left-down graph:

![Screenshot from 2022-08-10 15-29-03](https://user-images.githubusercontent.com/32936898/183843018-c063a287-5eed-403b-b157-6cca12067cd2.png)

In windows terminal/cmd, `btm` can work normally.

In conclusion, many programs cannot work normally in mintty/msys shell but can work normally in windows terminal/msys shell. The exception is for that btm, except fullscreen, it still exists other problems which windows terminal cannot fix. So maybe it should be reported to windows terminal?
